ddl/ddl_api.go

```diff
@@ -2512,8 +2512,13 @@ func processColumnOptions(ctx sessionctx.Context, col *table.Column, options []*
 			for _, colName := range findColumnNamesInExpr(opt.Expr) {
 				col.Dependences[colName.Name.L] = struct{}{}
 			}
+		case ast.ColumnOptionCollate:
+			col.Collate = opt.StrValue
+		case ast.ColumnOptionReference:
+			return errors.Trace(errUnsupportedModifyColumn.GenWithStackByArgs("with references"))
+		case ast.ColumnOptionFulltext:
+			return errors.Trace(errUnsupportedModifyColumn.GenWithStackByArgs("with full text"))
 		default:
-			// TODO: Support other types.
 			return errors.Trace(errUnsupportedModifyColumn.GenWithStackByArgs(opt.Tp))
```

Comment:
The error message is `%!s(ast.ColumnOptionType=12)`.
It seems the type of  `opt.Tp` is `ColumnOptionType` instead of `String`. I think we need to update this code.

Reply:
The `default` clause is impossible to reach.

Reply:
Currently, the `default` clause is impossible to reach. Could we just remove it?

Reply:
If there are new options added later, is it better to have this `default`?

Reply:
Updated.